src/main/java/redis/clients/jedis/ClientSideCache.java

```diff
@@ -0,0 +1,58 @@
+package redis.clients.jedis;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import redis.clients.jedis.exceptions.JedisException;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class ClientSideCache {
```

Comment:
Do you really need to implement your own cache storage? Don't you have something ready to use, maybe some lightweight library that provides a cache storage API?

I'm pretty much sure it will be quite time consuming to implement search mechanism in cache to search for all the data associated with key, as well as provide metadata for telemetry (cache hits, cache misses). Also, you have to think about TTL mechanism.

Reply:
We should define our cache APIs. I don't think we'll find anything that perfectly matches all our use cases. With a defined set of APIs, users may choose to provide their own implementation based on their preferred library.

At some point, this will be based on a library. But we have started simple.

Reply:
Internal separation of concern is a good idea